### PR TITLE
feat: security scans tab ui revamp

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
Revamp UI for scans listing
Clear url search filters on clear filters
add sorting in scan v1 modal

#Fixes https://github.com/devtron-labs/sprint-tasks/issues/674 https://github.com/devtron-labs/sprint-tasks/issues/699

Screenshot 2024-07-29 at 1 57 31 PM
Screenshot 2024-07-29 at 2 00 06 PM

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by AbhishekA1509 — COMMENTED** [review]
Please resolve sonar cloud comments

File: src/components/security/SecurityScansTab/SecurityScansTab.tsx [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        />
+    )
+
+    const renderScanList = () => {
+        if (scanListLoading) {
+            const arrayLoading = Array.from(Array(3)).map((index) => index)
+            return (
+                <div>
+                    {arrayLoading.map(() => (
+                        <div className="display-grid table__row-grid show-shimmer-loading dc__gap-16 px-20 py-10">
```
Comment (AbhishekA1509) [inline src/components/security/SecurityScansTab/SecurityScansTab.tsx:271]: Please add key

File: src/components/security/SecurityScansTab/SecurityScansTab.tsx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+                            <span className="child child-shimmer-loading" />
+                            <span className="child child-shimmer-loading w-250" />
+                        </div>
+                    ))}
+                </div>
+            )
+        }
+
+        if (!securityScansResult.result.securityScans.length) {
+            if (searchKey || severity.length || cluster.length || environment.length) {
```
Comment (AbhishekA1509) [inline src/components/security/SecurityScansTab/SecurityScansTab.tsx:284]: can create a variable areFiltersActive

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/components/security/SecurityScansTab/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    [SecurityScansTabSingleFilterKeys.searchType]:
+        searchParams.get(SecurityScansTabSingleFilterKeys.searchType) || 'appName',
+})
+
+export const getSearchLabelFromValue = (searchType: string) => {
+    if (searchType === SearchType.VULNERABILITY) return 'Vulnerability'
+    return 'Application'
+}
+
+export const getSeverityWithCount = (severityCount: SeverityCount) => {
```
[the remark this answers is not in the document]
Comment (arunjaindev) [reply src/components/security/SecurityScansTab/utils.tsx]: will pick later

File: src/components/security/SecurityScansTab/constants.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    name: '',
+    uniqueId: {
+        imageScanDeployInfoId: 0,
+        appId: 0,
+        envId: 0,
+    },
+}
+
+export const SEARCH_TYPE_OPTIONS = [
+    { label: 'Application', value: 'appName' },
```
Comment (AbhishekA1509) [inline src/components/security/SecurityScansTab/constants.tsx:13]: I think we have a enum for this value

File: src/components/security/SecurityScansTab/SecurityScansTab.tsx [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+                            options={SEARCH_TYPE_OPTIONS}
+                            classNamePrefix="search-type__select-picker"
+                            inputId="search-type__select-picker"
+                            name="search-type__select-picker"
+                            size={ComponentSizeType.large}
+                            onChange={updateSearchType}
+                        />
+                    </div>
+                    <SearchBar
+                        containerClassName="flex-grow-1"
```
Comment (AbhishekA1509) [inline src/components/security/SecurityScansTab/SecurityScansTab.tsx:212]: I think we should add max-width

File: src/components/security/SecurityScansTab/SecurityScansTab.tsx [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+                appId: scan.appId,
+                envId: scan.envId,
+            },
+        })
+    }
+
+    const renderHeader = () => (
+        <div className="table__row-grid display-grid dc__align-items-center dc__border-bottom dc__gap-16 px-20 w-100-imp py-4 dc__position-sticky dc__top-77 bcn-0">
+            <span className="icon-dim-24" />
+            <span className="fs-12 lh-20 fw-6 cn-7">
```
Comment (AbhishekA1509) [inline src/components/security/SecurityScansTab/SecurityScansTab.tsx:162]: Ideally should be a div

File: src/components/security/SecurityScansTab/SecurityScansTab.tsx [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+                <SortableTableHeaderCell
+                    title="APP NAME"
+                    isSorted={sortBy === SecurityListSortableKeys.APP_NAME}
+                    sortOrder={sortOrder}
+                    isSortable
+                    disabled={false}
+                    triggerSorting={handleAppNameSorting}
+                />
+            </span>
+            <span className="fs-12 lh-20 fw-6 cn-7">
```
Comment (AbhishekA1509) [inline src/components/security/SecurityScansTab/SecurityScansTab.tsx:172]: Same here

File: src/components/security/SecurityScansTab/SecurityScansTab.tsx [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+    const [scanListLoading, securityScansResult, scanListError] = useAsync(
+        () =>
+            abortPreviousRequests(
+                () => getSecurityScanList(payload, abortControllerRef.current.signal),
+                abortControllerRef,
+            ),
+        [filterConfig],
+    )
+
+    if (!scanListLoading && scanListError) {
```
Comment (AbhishekA1509) [inline src/components/security/SecurityScansTab/SecurityScansTab.tsx:119]: We need getIsRequestAborted as well here, please test those cases as well

File: src/components/security/SecurityScansTab/SecurityScansTab.tsx [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+            getFormattedValue={getLabelFromValue}
+            onRemoveFilter={updateSearchParams}
+            clearFilters={clearFilters}
+            className="w-100 pb-12-imp pt-0-imp px-20"
+            clearButtonClassName="dc__no-background-imp dc__no-border-imp dc__tab-focus"
+        />
+    )
+
+    const renderScanList = () => {
+        if (scanListLoading) {
```
Comment (AbhishekA1509) [inline src/components/security/SecurityScansTab/SecurityScansTab.tsx:266]: need || getIsRequestAborted(error)

File: src/services/service.ts [hunk 10]
```diff
@@ -273,11 +273,17 @@ function parseLastExecutionResponse(response): LastExecutionResponseType {
     const critical = vulnerabilities
```
[the remark this answers is not in the document]
Comment (AbhishekA1509) [reply src/services/service.ts]: .

[the remark this answers is not in the document]
Comment (arunjaindev) [reply src/services/service.ts]: parsing is not same, some keys are missing in fe-common

**Review by AbhishekA1509 — APPROVED** [review]

**Review by AbhishekA1509 — APPROVED** [review]
Please add type on SEARCH_TYPE_OPTIONS as well

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**